p2p/src/protocol.rs

```diff
@@ -1098,7 +1098,11 @@ impl<T: BlockTree, F: Filters, P: peer::Store> traits::Protocol for Protocol<T,
                 }
                 Command::Rescan { from, to, watch } => {
                     debug!(target: self.target, "Received command: Rescan({:?}, {:?})", from, to);
-                    self.cbfmgr.rescan(from, to, watch, &self.tree);
+
+                    // A rescan with a new watch list may return matches on cached filters.
+                    for hash in self.cbfmgr.rescan(from, to, watch, &self.tree) {
```

Comment:
mh sorry for the question here, but this looks like magic for me. How the following call can work in a if it is a void function?

`self.cbfmgr.rescan(from, to, watch, &self.tree)`

P.S: Stupid question I know!

Reply:
It is no longer void with this changeset, since `rescan` immediately checks scripts against cached filters!

Reply:
Oh yes, the magic is gone, thanks for the clarification